packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -212,6 +209,49 @@ export function BlockSettingsDropdown( {
 					className="block-editor-block-settings-menu"
 					popoverProps={ POPOVER_PROPS }
 					noIcons
+					menuProps={ {
+						/**
+						 * @param {KeyboardEvent} event
+						 */
+						onKeyDown( event ) {
+							if ( event.defaultPrevented ) return;
+
+							if (
+								isMatch( 'core/block-editor/remove', event ) &&
+								canRemove
+							) {
+								event.preventDefault();
+								updateSelectionAfterRemove( onRemove() );
```

Comment:
`updateSelectionAfterRemove` doesn't seem to be expecting any arguments — why do we pass `onRemove()` ? (same would apply for the instances of `updateSelectionAfterRemove` in this function)